pkg/cmd/init/exec.go

```diff
@@ -76,6 +76,11 @@ func (o *Options) validate() error {
 		return fmt.Errorf("registry should not be empty")
 	}
 
+	// If --wait is set, some information during initialize process will print to output, the output would not keep
+	// machine readable, so this behavior should be disabled
+	if o.wait && o.output != "text" {
+		return fmt.Errorf("output should be text if --wait is set")
+	}
```

Comment:
Disabling json format when using --wait is not great, but good enough.

A program can use

    clusteradm init --wait ...

To initialize, dropping or logging the text, and then use:

    clusteradm get token --output json

To fetch the hub info.

We can improve this later in several ways:
- Don't log anything progress messages when using json
- Or use jsonlines[1] format, so all lines are valid json

Possible json output:

    {"phase": "Progressing", "description": "Wait  for ... crd to be ready"}
    {"phase": "Progressing", "description": "Wait for ... crd to be established"}
    {"phase": "Progressing", "description": "CRD successfully registered"}
    {"phase: "Completed", "description": "...", "hub-token": ..., "hub-apiserver": ...}

The program reading this will read one line at a time, parsing the json. It can 
log the progress messages if it wants. It will take the hub-token and hub-apiserver
from the last message with the "Completed" phase.

In ovirt-img we use jsonlines to provide machine readable progress[2].
    
[1] https://jsonlines.org/
[2] https://github.com/oVirt/ovirt-imageio/pull/160